cpu/stm32f0/periph/gpio.c

```diff
@@ -24,315 +24,126 @@
+ */
 
-    RCC->AHBENR |= (1 << gpio_clock_map[dev]);
+static inline GPIO_TypeDef *_port(gpio_t pin)
+{
+    return (GPIO_TypeDef *)(pin & ~(0x0f));
+}
 
-    port->PUPDR &= ~(3 << (2 * pin));           /* configure push-pull resistors */
-    port->PUPDR |= (pullup << (2 * pin));
+/**
+ * @brief   Extract the port number form the given identifier
+ *
+ * The port number is extracted by looking at bits 10, 11, 12, 13 of the base
+ * register addresses.
+ */
+static inline int _port_num(gpio_t pin)
+{
+    return ((pin >> 10) & 0x0f);
+}
+
+/**
+ * @brief   Extract the pin number from the last 4 bit of the pin identifier
+ */
+static inline int _pin_num(gpio_t pin)
+{
+    return (pin & 0x0f);
+}
 
+int gpio_init(gpio_t pin, gpio_dir_t dir, gpio_pp_t pullup)
+{
+    GPIO_TypeDef *port = _port(pin);
+    int pin_num = _pin_num(pin);
+
+    /* enable clock */
+    RCC->AHBENR |= (RCC_AHBENR_GPIOAEN << _port_num(pin));
+    /* configure pull register */
+    port->PUPDR &= ~(3 << (2 * pin_num));
+    port->PUPDR |= (pushpull << (2 * pin_num));
+    /* set direction */
     if (dir == GPIO_DIR_OUT) {
-        port->MODER &= ~(2 << (2 * pin));           /* set pin to output mode */
-        port->MODER |= (1 << (2 * pin));
-        port->OTYPER &= ~(1 << pin);                /* set to push-pull configuration */
-        port->OSPEEDR |= (3 << (2 * pin));          /* set to high speed */
-        port->ODR &= ~(1 << pin);                   /* set pin to low signal */
+        port->MODER &= ~(3 << (2 * pin_num));   /* set pin to output mode */
+        port->MODER |= (1 << (2 * pin_num));
+        port->OTYPER &= ~(1 << pin_num);        /* set to push-pull */
+        port->OSPEEDR |= (3 << (2 * pin_num));  /* set to high speed */
+        port->ODR &= ~(1 << pin_num);           /* set pin to low signal */
     }
     else {
-        port->MODER &= ~(3 << (2 * pin));           /* configure pin as input */
-
+        port->MODER &= ~(3 << (2 * pin_num));   /* configure pin as input */
     }
-
-    return 0; /* all OK */
+    return 0;
 }
 
-int gpio_init_int(gpio_t dev, gpio_pp_t pullup, gpio_flank_t flank, gpio_cb_t cb, void *arg)
+int gpio_init_int(gpio_t pin, gpio_pp_t pullup, gpio_flank_t flank, gpio_cb_t cb, void *arg)
 {
-    int res;
-    uint8_t pin;
-
-    if (dev >= GPIO_NUMOF) {
-        return -1;
-    }
+    int pin_num = _pin_num(pin);
+    int port_num = _port_num(pin);
 
-    pin = gpio_pin_map[dev];
-
-    /* configure pin as input */
-    res = gpio_init(dev, GPIO_DIR_IN, pullup);
-    if (res < 0) {
-        return res;
-    }
+    /* set callback */
+    exti_ctx[pin_num].cb = cb;
+    exti_ctx[pin_num].arg = arg;
 
-    /* set interrupt priority (its the same for all EXTI interrupts) */
-    NVIC_SetPriority(EXTI0_1_IRQn, GPIO_IRQ_PRIO);
-    NVIC_SetPriority(EXTI2_3_IRQn, GPIO_IRQ_PRIO);
-    NVIC_SetPriority(EXTI4_15_IRQn, GPIO_IRQ_PRIO);
+    /* enable the SYSCFG clock */
+    RCC->APB2ENR |= RCC_APB2ENR_SYSCFGEN;
```

Comment:
same as line line 105, so you can remove this. Quote from the CPUs header file: "/* Old Bit definition maintained for legacy purpose */"....